node/src/runner.rs

```diff
@@ -104,16 +104,19 @@ pub async fn initialize(
     );
 
     // RPC subsystem
-    let _rpc = manager.add_subsystem(
-        "rpc",
-        rpc::Builder::new(node_config.rpc)
-            .register(chainstate.clone().into_rpc())
-            .register(mempool.into_rpc())
-            .register(NodeRpc::new(manager.make_shutdown_trigger()).into_rpc())
-            .register(p2p.clone().into_rpc())
-            .build()
-            .await?,
-    );
+    if node_config.rpc.http_enabled.unwrap_or(true) || node_config.rpc.ws_enabled.unwrap_or(true) {
+        // TODO: get rid of the unwrap_or() after fixing the issue in #446
+        let _rpc = manager.add_subsystem(
+            "rpc",
+            rpc::Builder::new(node_config.rpc)
+                .register(chainstate.clone().into_rpc())
+                .register(mempool.into_rpc())
+                .register(NodeRpc::new(manager.make_shutdown_trigger()).into_rpc())
+                .register(p2p.clone().into_rpc())
+                .build()
+                .await?,
+        );
+    }
```

Comment:
Would it make sense for `rpc-http` and `rpc-ws` to be separate subsystems and only start the ones that are enabled? The `rpc` crate now has way too many pattern matches on `Option` to my liking.

After some refactoring in the `rpc` module, this bit could look something like:

```rust
let rpc_builder = rpc::builder::new().register(/*...*/).register(/*...*/);
if node_config.rpc.http_enabled() {
    manager.add_subsystem("rpc-http", rpc_builder.build_http(node_config.rpc.http_addr()).await);
}
if node_config.rpc.ws_enabled() {
    manager.add_subsystem("rpc-ws", rpc_builder.build_ws(node_config.rpc.ws_addr()).await);
}
```

Reply:
They both use the same function objects, so we'll have to duplicate the handlers. Not really the best choice.